src/node_process.cc

```diff
@@ -768,10 +773,9 @@ void EnvEnumerator(const PropertyCallbackInfo<Array>& info) {
     }
     const uint16_t* two_byte_buffer = reinterpret_cast<const uint16_t*>(p);
     const size_t two_byte_buffer_len = s - p;
-    argv[idx] = String::NewFromTwoByte(isolate,
-                                       two_byte_buffer,
-                                       String::kNormalString,
-                                       two_byte_buffer_len);
+    v8::MaybeLocal<String> rc = String::NewFromTwoByte(
+        isolate, two_byte_buffer, NewStringType::kNormal, two_byte_buffer_len);
+    argv[idx] = rc.ToLocalChecked();
```

Comment:
@refack That's what I thought ! But i'm not sure what i'm supposed to put on the statement if IsEmpty() return true.

Do i have to continue the while or return ?

Reply:
IIUC having `rc.IsEmpty()` mean something went wrong and there is an exception waiting to be thrown when the code returns to JS land.
So you just `return`.

Reply:
Right now, this can only happen when the string exceeds the maximum length supported by the engine. In practice, that won’t happen for environment variables, so checking `IsEmpty()` is optional.

If you do decide to do it, there’s a catch:

The `MaybeLocal<String>` case is a bit odd – currently, V8 does *not* throw an exception when it return an empty `MaybeLocal` from the `String` creation methods, which is unlike almost all `Maybe<>`/`MaybeLocal<>` usage it has. (They only have a `TODO` comment on this matter atm.)

You could use `isolate->ThrowException(ERR_STRING_TOO_LONG(isolate));` from `node_errors.h` to make us throw an exception in this case.

Reply:
oh ok, i understand ! Thanks, i work on it.